### ISSUE TITLE
add internal OrderBy / OrderByDescending for IRedisCollection<T>

### ISSUE THREAD

**Issue by VagyokC4** [redis/redis-om-dotnet#103]
Can you please define IRedisCollection< T> OrderBy< T>() and IRedisCollection< T> OrderByDescending< T>()

I noticed this behavior before you wired up the async commands.  Before you wired them up internally, my code was trying to use Linq / EF code to perform the functions.  After you wired up internally, they were found and my ambiguous reference went away.

So this works no problem:
![image](https://user-images.githubusercontent.com/253250/165643411-a573b792-8950-4972-bd3f-503b46269036.png)

But once I try to add sorting to the query, I get errors:

![image](https://user-images.githubusercontent.com/253250/165643498-1b6b1071-3f8c-42fc-8f9b-ef0fb80085cb.png)

saying:

![image](https://user-images.githubusercontent.com/253250/165643539-cb6f872e-a9b5-453f-ad29-9d7c292826ad.png)

In the code base I see you have them wired up for 
- GroupedAggregationSet< T>
- RedisAggregationSet< T>

Just not for
- IRedisCollection< T>

**Comment by VagyokC4** [comment]
Hi @slorello89, 

Just pinging you on this issue here.

**Comment by slorello89** [comment]
Hey @VagyokC4 

Sorry I've been a bit heads-down the last couple weeks ( and will be for the next couple) but Looking at the example you provided, the only issue I can see is that ToListAsync isn't going to work on the IOrderedEnumerable you get back from The OrderBy/OrderByDescending. You should just be able to cast it to an `IRedisCollection<T>` to get it to work correctly. The ambiguous reference you see isn't something I can readily reproduce - maybe something to do with your imports?

**Comment by VagyokC4** [comment]
> Sorry I've been a bit heads-down the last couple weeks ( and will be for the next couple) but Looking at the example you provided, the only issue I can see is that ToListAsync isn't going to work on the IOrderedEnumerable you get back from The OrderBy/OrderByDescending. You should just be able to cast it to an `IRedisCollection<T>` to get it to work correctly. The ambiguous reference you see isn't something I can readily reproduce - maybe something to do with your imports?

Yeah, I kind of figured as much.  Godspeed with your work.  Like I said, I was having the issue before when I was doing my Async testing, but all those went away as soon as you brought reworked the sync/async commands.   As soon as they were located inside redis.om, those ambigious calls when away, as my IDE allowed me to reference the redis-om version.

Can you give me an example of what that casting would look like?

**Comment by slorello89** [comment]
```
var result = await ((IRedisCollection<Person>) collection.Where(x=>x.TagField=="hello").OrderBy(x => x.Age).Skip(0).Take(10)).ToListAsync();
```

**Comment by VagyokC4** [comment]
> var result = await ((IRedisCollection<Person>) collection.Where(x=>x.TagField=="hello").OrderBy(x => x.Age).Skip(0).Take(10)).ToListAsync();

So the top was my attempt, the bottom was your example.
![image](https://user-images.githubusercontent.com/253250/166002775-a7cb5b0b-9b8c-425d-b6ce-ea07f0d50247.png)

I'm still getting this on my example:
![image](https://user-images.githubusercontent.com/253250/166003069-b09f509d-1bb8-4e4b-98e4-f04ff560c737.png)

In addition, I get this on your example
![image](https://user-images.githubusercontent.com/253250/166003270-2670bfda-b352-4a27-9cb6-c8cb090db38f.png)

Any other ideas?

**Comment by slorello89** [comment]
Try adding:

```
using System.Linq;
```

to your imports?

**Comment by VagyokC4** [comment]
> Try adding:
> 
> ```
> using System.Linq;
> ```
> 
> to your imports?

Yup.  That was already there.

Going to AsQueryable(), then back to IRedisCollection<T> is at least a work around.

**Comment by slorello89** [comment]
Should be Fixed in #130
